### ISSUE TITLE
젯슨나노로 얼굴, 자세 판별모델구현

### ISSUE THREAD

**Issue by kim20s2** [won-jong-wan/Pride-and-Prejudice#31]
https://github.com/won-jong-wan/Pride-and-Prejudice/issues/15 -> jetson-inference(자세감지)
https://github.com/won-jong-wan/Pride-and-Prejudice/issues/22 -> jetson_zoo(자세감지, 얼굴인식 등)

yolov5 + RepVGG
https://github.com/George-Ogden/emotion
해당 모델 젯슨 나노로 구현 시도
-> YOLO 기반 얼굴검출 + RepVGG 감정분류 2스테이지 구조라서 나노에겐 무거워서 FPS가 낮고, 의존성도 데스크톱 기준이라 구현 안됨

-------------------------------------------------------------------------------------------------------------------------------------------------

### 얼굴 분류 모델 설명
백본: MobileNetV2 (depthwise separable conv + inverted residual).
헤드: GlobalAvgPool → (Dropout) → Fully-Connected(Linear) → Softmax.
클래스 수: 보통 7(angry, disgust, fear, happy, sad, surprise, neutral).

MobileNetV2 백본으로 PyTorch로 모델 만든 후 TensorRT로 변환하여 젯슨나노에서 7-class로 추론
-> 얼굴 표정 분류 모델 7-class로 구현 **완료** (얼굴 검출 -> 얼굴 크롭/전처리 -> 얼굴 분류)
-> 최대 25FPS 나옴
-> 학습 시 calib 이미지는 FER2013 참고(500장정도)
-> 학습 했을때 정확도 68%정도 나왔었음
-> 현재는 USB 웹캠(640x480) 으로만 구현완료
-> 자세 모델 추가하여 한번에 얼굴 표정 분류 + 자세 추정 해주는 걸 구현해볼 예정임
- 학습용 파이썬 파일(7-class)
[train_fer.py](https://github.com/user-attachments/files/22921387/train_fer.py)
- confusion matrix(data sets)
<img width="1200" height="1000" alt="Image" src="https://github.com/user-attachments/assets/43602cf9-65d5-44fa-b38d-38900a3cadd9" />

### 자세 모델 movenet 설명
백본: 경량 CNN 백본(공개 자료에 따르면 MobileNetV2 기반) + **FPN(Feature Pyramid Network)**로 다중 스케일 특징을 만듬
입력: 1×192×192×3 RGB, 정규화 이미지
출력: 1×1×17×3 (y, x, score) — COCO 17 keypoints의 정규화 좌표와 신뢰도 점수. 
FP16 변형: 가중치/연산을 float16으로 사용하여 메모리·연산량을 줄인 TFLite 변형(정밀도 약간 감소, 속도/메모리 이점)
포지셔닝/성능 특성: Lightning은 지연시간 최소화를 목표로 한 소형 모델(Thunder보다 작고 빠름). 

<img width="1024" height="673" alt="Image" src="https://github.com/user-attachments/assets/8f1fab91-d3a0-476b-866a-0afdeff035b5" />

참고: https://www.tensorflow.org/hub/tutorials/movenet?hl=ko

기존 얼굴 분류 모델 과 동시에 구현 진행
-> trt_utils.py (서브파일)과 run_fer_pose.py로 구현 **성공**
-> 실행 명령어 python3 run_fer_pose.py --src 1 --show_fps
-> 얼굴, 자세 동시 추정 시(모든 골격) 약 15FPS정도 나옴
-> 일단 웹캠(640x480)으로만 구현완료, imx219 Tegracam과 인텔 리얼센스 웹캠으로도 해보는 중

**Comment by kim20s2** [comment]
7-class로 할시 angry, disgust, fear, happy는 확실히 잘 분류함 -> 면접때 7클래스까진 필요없어서 3클래스(positive, neutral, negative)로 줄여서 다시 모델 compile 할예정. 
Positive: happy
Neutral: neutral + (기본) surprise
Negative: angry, disgust, fear, sad
